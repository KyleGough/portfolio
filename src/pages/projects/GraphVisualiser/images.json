[
  {
    "src": "/img/graph-algorithm-visualiser1.jpg",
    "caption": "Nearest Neighbour with 2-Opt"
  },
  {
    "src": "/img/graph-algorithm-visualiser2.jpg",
    "caption": "Randomly generated nodes"
  },
  {
    "src": "/img/graph-algorithm-visualiser3.jpg",
    "caption": "Algorithm information"
  },
  { "src": "/img/graph-algorithm-visualiser4.jpg", "caption": "Graham Scan" },
  { "src": "/img/graph-algorithm-visualiser5.jpg", "caption": "Random route" }
]
