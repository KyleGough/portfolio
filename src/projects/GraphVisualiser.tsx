import { useEffect } from 'react';
import Section from '../components/Section';
import Divider from '../components/Divider';
import Chip from '../components/Chip';
import Pagination from '../components/Pagination';
import ImageFigure from '../components/ImageFigure';
import Screenshots from '../components/Screenshots';

export default function GraphVisualiser() {

    useEffect(() => {
        document.title = 'Graph Algorithm Visualiser - Web App to Visualise Various Graph Algorithms';
    }, []);

    const images = [
        { src:'/img/graph-algorithm-visualiser1.jpg', caption:'Nearest Neighbour with 2-Opt' },
        { src:'/img/graph-algorithm-visualiser2.jpg', caption:'Randomly generated nodes' },
        { src:'/img/graph-algorithm-visualiser3.jpg', caption:'Algorithm information' },
        { src:'/img/graph-algorithm-visualiser4.jpg', caption:'Graham Scan' },
        { src:'/img/graph-algorithm-visualiser5.jpg', caption:'Random route' }
    ];

    return (
        <>
        <Section>
            <h1 className='project-title'>Graph Algorithm Visualiser</h1>
            <p className='text-link-hover my-4'><time dateTime='2017-04'>April 2017</time></p>
            <p className='mb-4 max-w-reading'>Web application which visualises the process of various graph algorithms on randomised nodes. The application can simulate Prim's and Kruskal's algorithm for constructing minimum spanning trees, Graham scan to construct convex hulls, Greedy algorithm to construct minimal matchings, and Nearest neighbour with 2-Opt to create Hamiltonian cycles.</p>
            <p className='mb-4'>
                <span>Explore this project on </span>
                <a className='text-link hover:text-link-hover focus:text-link-hover' href='https://github.com/KyleGough/graph-algorithm-visualiser' target='_blank' rel='noopener noreferrer'>GitHub</a>
            </p>

            <div className='flex flex-row flex-wrap items-center mt-8 gap-4'>
                <Chip name='JavaScript' />
                <Chip name='Materialize' />
            </div>
        </Section>

        <Divider />

        <ImageFigure image={images[0]} />

        <Divider />

        <Section>
            <h2 className='project-header'>Features</h2>
            <ul className='project-list'>
                <li>Visualisation of the graph, unvisited nodes, visited nodes and edges connecting them.</li>
                <li>Choice of low number to high number of nodes in the graph.</li>
                <li>Algorithm statistics such as: Tour length, Minimum Spanning Tree length, Convex Hull nodes.</li>
                <li><strong>Minimum Spanning Tree Algorithms</strong> - Prim's, Kruskal's</li>
                <li><strong>Hamiltonian Path</strong> - Nearest Neighbour, Nearest Neighbour with 2-Opt, Random Tour</li>
                <li><strong>Convex Hull Algorithms</strong> - Graham Scan</li>
                <li><strong>Minimum Matching</strong> - Greedy (variant of Kruskal's algorithm)</li>
            </ul>
        </Section>

        <Divider />

        <Screenshots images={images} />

        <Divider />
        
        <Pagination
            previousTitle='Cavern Minesweeper'
            previousLink='/projects/cavern-minesweeper'
            nextTitle='URL Shortener'
            nextLink='/projects/url-shortener'
        />
        </>
    );
}