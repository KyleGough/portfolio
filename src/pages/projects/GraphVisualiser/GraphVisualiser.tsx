import React, { useEffect } from 'react';

import { Chip } from '../../../components/Chip';
import { Divider } from '../../../components/Divider';
import { ImageFigure } from '../../../components/ImageFigure';
import { Pagination } from '../../../components/Pagination';
import { Screenshots } from '../../../components/Screenshots';
import { Section } from '../../../components/Section';
import images from './images.json';

export const GraphVisualiser: React.FC = () => {
  useEffect(() => {
    document.title =
      'Graph Algorithm Visualiser - Web App to Visualise Various Graph Algorithms';
  }, []);

  return (
    <>
      <Section>
        <h1 className="project-title">Graph Algorithm Visualiser</h1>
        <p className="text-link-hover my-4">
          <time dateTime="2017-04">April 2017</time>
        </p>
        <p className="mb-4 max-w-reading">
          Web application which visualises the process of various graph
          algorithms on randomised nodes. The application can simulate
          Prim&apos;s and Kruskal&apos;s algorithm for constructing minimum
          spanning trees, Graham scan to construct convex hulls, Greedy
          algorithm to construct minimal matchings, and Nearest neighbour with
          2-Opt to create Hamiltonian cycles.
        </p>
        <p className="mb-4">
          <span>Explore this project on </span>
          <a
            className="text-link hover:text-link-hover focus:text-link-hover"
            href="https://github.com/KyleGough/graph-algorithm-visualiser"
            target="_blank"
            rel="noopener noreferrer"
          >
            GitHub
          </a>
        </p>

        <div className="flex flex-row flex-wrap items-center mt-8 gap-4">
          <Chip name="JavaScript" />
          <Chip name="Materialize" />
        </div>
      </Section>

      <Divider />

      <ImageFigure image={images[0]} />

      <Divider />

      <Section>
        <h2 className="project-header">Features</h2>
        <ul className="project-list">
          <li>
            Visualisation of the graph, unvisited nodes, visited nodes and edges
            connecting them.
          </li>
          <li>Choice of low number to high number of nodes in the graph.</li>
          <li>
            Algorithm statistics such as: Tour length, Minimum Spanning Tree
            length, Convex Hull nodes.
          </li>
          <li>
            <strong>Minimum Spanning Tree Algorithms</strong> - Prim&apos;s,
            Kruskal&apos;s
          </li>
          <li>
            <strong>Hamiltonian Path</strong> - Nearest Neighbour, Nearest
            Neighbour with 2-Opt, Random Tour
          </li>
          <li>
            <strong>Convex Hull Algorithms</strong> - Graham Scan
          </li>
          <li>
            <strong>Minimum Matching</strong> - Greedy (variant of
            Kruskal&apos;s algorithm)
          </li>
        </ul>
      </Section>

      <Divider />

      <Screenshots images={images} />

      <Divider />

      <Pagination
        previousTitle="Cavern Minesweeper"
        previousLink="/projects/cavern-minesweeper"
        nextTitle="URL Shortener"
        nextLink="/projects/url-shortener"
      />
    </>
  );
};
