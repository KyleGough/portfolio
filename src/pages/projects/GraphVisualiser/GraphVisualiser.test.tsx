import { render } from '@testing-library/react';
import React from 'react';
import { BrowserRouter } from 'react-router-dom';

import { GraphVisualiser } from './GraphVisualiser';

describe('GraphVisualiser project page', () => {
  it('renders', async () => {
    const { container } = render(
      <BrowserRouter>
        <GraphVisualiser />
      </BrowserRouter>
    );
    expect(container).toMatchSnapshot();
  });
});
