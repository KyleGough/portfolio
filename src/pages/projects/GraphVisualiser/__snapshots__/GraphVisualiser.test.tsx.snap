// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GraphVisualiser project page renders 1`] = `
<div>
  <div
    class="first:pt-24 container text-primary py-8 last:pb-16"
  >
    <h1
      class="project-title"
    >
      Graph Algorithm Visualiser
    </h1>
    <p
      class="text-link-hover my-4"
    >
      <time
        datetime="2017-04"
      >
        April 2017
      </time>
    </p>
    <p
      class="mb-4 max-w-reading"
    >
      Web application which visualises the process of various graph algorithms on randomised nodes. The application can simulate Prim's and Kruskal's algorithm for constructing minimum spanning trees, Graham scan to construct convex hulls, Greedy algorithm to construct minimal matchings, and Nearest neighbour with 2-Opt to create Hamiltonian cycles.
    </p>
    <ul
      class="project-list"
    >
      <li
        class="mb-4"
      >
        <span>
          Explore this project on 
        </span>
        <a
          class="text-link hover:text-link-hover focus:text-link-hover"
          href="https://github.com/KyleGough/graph-algorithm-visualiser"
          rel="noopener noreferrer"
          target="_blank"
        >
          GitHub
        </a>
      </li>
      <li
        class="mb-4"
      >
        <span>
          Explore this project live 
        </span>
        <a
          class="text-link hover:text-link-hover focus:text-link-hover"
          href="https://kylegough.github.io/graph-algorithm-visualiser/"
          rel="noopener noreferrer"
          target="_blank"
        >
          here
        </a>
      </li>
    </ul>
    <div
      class="flex flex-row flex-wrap items-center mt-8 gap-4"
    >
      <div
        class="text-chip bg-chip-light text-xs px-4 py-0.5 h-8 leading-chip rounded-2xl font-extrabold"
      >
        JavaScript
      </div>
      <div
        class="text-chip bg-chip-light text-xs px-4 py-0.5 h-8 leading-chip rounded-2xl font-extrabold"
      >
        Materialize
      </div>
    </div>
  </div>
  <hr
    class="m-8 h-px bg-divider"
  />
  <div
    class="first:pt-24 container text-primary py-8 last:pb-16"
  >
    <img
      alt="Nearest Neighbour with 2-Opt"
      class="border-2 border-white rounded-lg text-center items-center shadow mx-auto"
      src="/img/graph-algorithm-visualiser1.jpg"
    />
    <p
      class="mt-4 mb-2 text-sm text-center text-light font-thin"
    >
      Nearest Neighbour with 2-Opt
    </p>
  </div>
  <hr
    class="m-8 h-px bg-divider"
  />
  <div
    class="first:pt-24 container text-primary py-8 last:pb-16"
  >
    <h2
      class="project-header"
    >
      Features
    </h2>
    <ul
      class="project-list"
    >
      <li>
        Visualisation of the graph, unvisited nodes, visited nodes and edges connecting them.
      </li>
      <li>
        Choice of low number to high number of nodes in the graph.
      </li>
      <li>
        Algorithm statistics such as: Tour length, Minimum Spanning Tree length, Convex Hull nodes.
      </li>
      <li>
        <strong>
          Minimum Spanning Tree Algorithms
        </strong>
         - Prim's, Kruskal's
      </li>
      <li>
        <strong>
          Hamiltonian Path
        </strong>
         - Nearest Neighbour, Nearest Neighbour with 2-Opt, Random Tour
      </li>
      <li>
        <strong>
          Convex Hull Algorithms
        </strong>
         - Graham Scan
      </li>
      <li>
        <strong>
          Minimum Matching
        </strong>
         - Greedy (variant of Kruskal's algorithm)
      </li>
    </ul>
  </div>
  <hr
    class="m-8 h-px bg-divider"
  />
  <div
    class="first:pt-24 container text-primary py-8 last:pb-16"
  >
    <h2
      class="text-3xl mb-12"
    >
      Screenshots
    </h2>
    <div
      class="grid grid-cols-12 gap-4"
    >
      <figure
        class="col-span-12 md:col-span-6 lg:col-span-4"
      >
        <img
          alt="Nearest Neighbour with 2-Opt"
          class="cursor-pointer border-2 border-link hover:border-link-hover"
          loading="lazy"
          src="/img/graph-algorithm-visualiser1.jpg"
        />
        <figcaption
          class="my-2 text-sm text-center text-light font-thin"
        >
          Nearest Neighbour with 2-Opt
        </figcaption>
      </figure>
      <figure
        class="col-span-12 md:col-span-6 lg:col-span-4"
      >
        <img
          alt="Randomly generated nodes"
          class="cursor-pointer border-2 border-link hover:border-link-hover"
          loading="lazy"
          src="/img/graph-algorithm-visualiser2.jpg"
        />
        <figcaption
          class="my-2 text-sm text-center text-light font-thin"
        >
          Randomly generated nodes
        </figcaption>
      </figure>
      <figure
        class="col-span-12 md:col-span-6 lg:col-span-4"
      >
        <img
          alt="Algorithm information"
          class="cursor-pointer border-2 border-link hover:border-link-hover"
          loading="lazy"
          src="/img/graph-algorithm-visualiser3.jpg"
        />
        <figcaption
          class="my-2 text-sm text-center text-light font-thin"
        >
          Algorithm information
        </figcaption>
      </figure>
      <figure
        class="col-span-12 md:col-span-6 lg:col-span-4"
      >
        <img
          alt="Graham Scan"
          class="cursor-pointer border-2 border-link hover:border-link-hover"
          loading="lazy"
          src="/img/graph-algorithm-visualiser4.jpg"
        />
        <figcaption
          class="my-2 text-sm text-center text-light font-thin"
        >
          Graham Scan
        </figcaption>
      </figure>
      <figure
        class="col-span-12 md:col-span-6 lg:col-span-4"
      >
        <img
          alt="Random route"
          class="cursor-pointer border-2 border-link hover:border-link-hover"
          loading="lazy"
          src="/img/graph-algorithm-visualiser5.jpg"
        />
        <figcaption
          class="my-2 text-sm text-center text-light font-thin"
        >
          Random route
        </figcaption>
      </figure>
    </div>
  </div>
  <hr
    class="m-8 h-px bg-divider"
  />
  <div
    class="first:pt-24 container text-primary py-8 last:pb-16"
  >
    <div
      class="flex justify-between"
    >
      <a
        class="flex items-center fill-link text-link hover:fill-link-hover hover:text-link-hover"
        href="/projects/cavern-minesweeper"
      >
        <svg
          class="w-6 h-6"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M0 0h24v24H0V0z"
            fill="none"
            opacity=".87"
          />
          <path
            d="M17.51 3.87L15.73 2.1 5.84 12l9.9 9.9 1.77-1.77L9.38 12l8.13-8.13z"
          />
        </svg>
        <div
          class="text-center ml-4"
        >
          <p
            class="text-sm"
          >
            Previous Project
          </p>
          <p>
            Cavern Minesweeper
          </p>
        </div>
      </a>
      <a
        class="flex items-center fill-link text-link hover:fill-link-hover hover:text-link-hover"
        href="/projects/url-shortener"
      >
        <div
          class="text-center mr-4"
        >
          <p
            class="text-sm"
          >
            Next Project
          </p>
          <p>
            URL Shortener
          </p>
        </div>
        <svg
          class="w-6 h-6"
          enable-background="new 0 0 24 24"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <g>
            <path
              d="M0,0h24v24H0V0z"
              fill="none"
            />
          </g>
          <g>
            <polygon
              points="6.23,20.23 8,22 18,12 8,2 6.23,3.77 14.46,12"
            />
          </g>
        </svg>
      </a>
    </div>
  </div>
</div>
`;
